pyk/src/pyk/prelude.py

```diff
@@ -60,29 +71,34 @@ def leInt(i1, i2):
     return KApply('_<=Int_', [i1, i2])
 
 
-def mlEquals(a1, a2):
-    return KApply('#Equals', [a1, a2])
+# TODO default sort K can be tightened using basic type inference
+def mlEquals(term1: KInner, term2: KInner, sort1: Union[str, KSort] = K, sort2: Union[str, KSort] = K) -> KApply:
+    return KApply(KLabel('#Equals', (sort1, sort2)), (term1, term2))
+
+
+def mlEqualsTrue(term: KInner) -> KApply:
+    return mlEquals(TRUE, term, BOOL)
 
 
-def mlEqualsTrue(b):
-    return mlEquals(TRUE, b)
+def mlTop(sort: Union[str, KSort] = K) -> KApply:
+    return KApply(KLabel('#Top', (sort,)))
 
 
-def mlTop():
-    return TOP
+def mlBottom(sort: Union[str, KSort] = K) -> KApply:
+    return KApply(KLabel('#Top', (sort,)))
```

Comment:
Should we name hardcoded sorts like `Sort_K` or `Sort_GENERATED_TOP_CELL`? To make them more distinct, and to allow also later `Label_GENERATED_TOP_CELL`.

Reply:
Yeah, that makes sense. We can also put them in a module `sorts` or in a class `Sorts`, and then have `sorts.K` or `Sorts.K`, repsectively.

Reply:
Oh that wolud be great. Could we have things like `Sorts.K` and `Labels.EmptyK` defined in `prelude.py`? It seems the most appropriate place for them.

Reply:
I agree, will move them there.

Reply:
Done.